nifi-api/src/main/java/org/apache/nifi/provenance/upload/FileResource.java

```diff
@@ -20,15 +20,26 @@
  * Holds information of a file resource for UPLOAD
  * provenance events.
  */
-public interface FileResource {
+public class FileResource {
```

Comment:
This is a breaking change, so it should be reverted.

Reply:
@exceptionfactory I included the FileResource interface in this pull request as an interface, but later realized it's unnecessary as it functions as a straightforward DTO. Moreover, if I introduce the StandardFileResource implementation to nifi-data-provenance-utils (the place I thought to add it), it will create a dependency on nifi processors.

Reply:
I can reduce the number of methods, I followed the signatures of the send method. I can check which ones are more frequently used and eliminate the others.

Reply:
Thanks for the clarification, I should have reviewed the full history, that makes sense.

Yes, I recommend defining the methods required for the common use cases.

Reply:
I excluded methods based on if they were either unused or less commonly utilized in the SEND events.